src/e2e/get_tickmap.rs

```diff
@@ -0,0 +1,493 @@
+
+        create_position!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            10,
+            20,
+            liquidity_delta,
+            pool.sqrt_price,
+            SqrtPrice::max_instance(),
+            alice
+        )
+        .unwrap();
+
+        let tickmap = get_tickmap!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            pool.current_tick_index,
+            alice
+        );
+
+        assert_eq!(
+            tickmap[0],
+            (346, 0b1100000000000000000000000000000000000000)
+        );
+        assert_eq!(tickmap.len(), 1);
+
+        Ok(())
+    }
+
+    #[ink_e2e::test]
+    async fn test_get_tickmap_edge_ticks_intialized(
+        mut client: ink_e2e::Client<C, E>,
+    ) -> E2EResult<()> {
+        let alice = ink_e2e::alice();
+        let dex = create_dex!(client, InvariantRef, Percentage::new(0));
+        let initial_amount = 10u128.pow(10);
+        let (token_x, token_y) = create_tokens!(client, TokenRef, initial_amount, initial_amount);
+
+        approve!(client, TokenRef, token_x, dex, initial_amount, alice).unwrap();
+        approve!(client, TokenRef, token_y, dex, initial_amount, alice).unwrap();
+
+        let fee_tier = FeeTier::new(Percentage::from_scale(5, 1), 1).unwrap();
+        let pool_key = PoolKey::new(token_x, token_y, fee_tier).unwrap();
+        let init_tick = 0;
+        let init_sqrt_price = calculate_sqrt_price(init_tick).unwrap();
+
+        add_fee_tier!(client, InvariantRef, dex, fee_tier, alice).unwrap();
+
+        let result = create_pool!(
+            client,
+            InvariantRef,
+            dex,
+            token_x,
+            token_y,
+            fee_tier,
+            init_sqrt_price,
+            init_tick,
+            alice
+        );
+        assert!(result.is_ok());
+
+        let pool = get_pool!(client, InvariantRef, dex, token_x, token_y, fee_tier).unwrap();
+
+        let liquidity_delta = Liquidity::new(1000);
+
+        create_position!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            -221818,
+            -221817,
+            liquidity_delta,
+            pool.sqrt_price,
+            SqrtPrice::max_instance(),
+            alice
+        )
+        .unwrap();
+
+        create_position!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            221817,
+            221818,
+            liquidity_delta,
+            pool.sqrt_price,
+            SqrtPrice::max_instance(),
+            alice
+        )
+        .unwrap();
+
+        let tickmap = get_tickmap!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            pool.current_tick_index,
+            alice
+        );
+
+        assert_eq!(tickmap[0], (0, 0b11));
+        assert_eq!(
+            tickmap[1],
+            (
+                6931,
```

Comment:
Here, you can easily validate the `get_max_chunk` function. The index returned by `get_tickmap` should match the invocation of `get_max_chunk`.